lib/TwoFactorAuth.php

```diff
@@ -194,16 +189,10 @@ public function getRngProvider(): IRNGProvider
         throw new TwoFactorAuthException('Unable to find a suited RNGProvider');
     }
 
-    /**
-     * @throws TwoFactorAuthException
-     */
     public function getTimeProvider(): ITimeProvider
     {
         // Set default time provider if none was specified
-        if (null === $this->timeprovider) {
-            return $this->timeprovider = new LocalMachineTimeProvider();
-        }
-        return $this->timeprovider;
+        return $this->timeprovider ?? ($this->timeprovider = new LocalMachineTimeProvider());
```

Comment:
isn't this what ??= is for? (Null Coalescing Assignment)

Reply:
Good call. Fixed in https://github.com/RobThree/TwoFactorAuth/pull/108/commits/9bc454f42501a89ce98860e1e9b7b774e555bf38.